src/mqt/qecc/circuit_compilation/code_switching_compiler.py

```
@@ -0,0 +1,234 @@
+
+        Returns:
+        -------
+        str
+            The unique node identifier created for this operation.
+        """
+        node_id = f"{gate_type}_q{qubit}_d{depth}"
+        self.G.add_node(node_id, gate=gate_type, qubit=qubit, depth=depth)
+        return node_id
+
+    def add_edge_with_capacity(
+        self, u: str, v: str, capacity: float, edge_type: str = "temporal", bidirectional: bool = True
```

Ruff (0.14.3)

Boolean-typed positional argument in function definition

(FBT001)